.travis.yml

```diff
@@ -22,104 +27,521 @@ cache:
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.4"
+     script: *some_dependencies
+
+   - stage: some_dependencies
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.5"
+     script: *some_dependencies
+
+   - stage: dependencies
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.4.2"
+     script: *dependencies
+
+   - stage: dependencies
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.4.3"
+     script: *dependencies
+
+   - stage: dependencies
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.4.4"
+     script: *dependencies
+
+   - stage: dependencies
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.1"
+     script: *dependencies
+
+   - stage: dependencies
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.2"
+     script: *dependencies
+
+   - stage: dependencies
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.3"
+     script: *dependencies
+
+   - stage: dependencies
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.4"
+     script: *dependencies
+
+   - stage: dependencies
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.5"
+     script: *dependencies
+
+   - stage: build
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.4.2"
+     script: *build
+
+   - stage: build
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.4.3"
+     script: *build
+
+   - stage: build
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.4.4"
+     script: *build
+
+   - stage: build
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.1"
+     script: *build
+
+   - stage: build
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.2"
+     script: *build
+
+   - stage: build
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.3"
+     script: *build
+
+   - stage: build
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.4"
+     script: *build
+
+   - stage: build
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.5"
+     script: *build
+
+   - stage: test
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.4.2"
+     script: *test
+
+   - stage: test
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.4.3"
+     script: *test
+
+   - stage: test
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.4.4"
+     script: *test
+
+   - stage: test
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.1"
+     script: *test
+
+   - stage: test
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.2"
+     script: *test
+
+   - stage: test
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.3"
+     script: *test
+
+   - stage: test
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.4"
+     script: *test
+
+   - stage: test
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.5"
+     script: *test
```

Comment:
Is there a way to specify a matrix for the GHC_VER env?

Reply:
I don't think so.  Would have been nice.  Seems that stages and a matrix of environment variables cannot be used together:
https://github.com/peterbecich/haskell-ide-engine/blob/aadbea8eb9d1212dc3752ac5a843469d5f42cd94/.travis.yml
https://travis-ci.org/peterbecich/haskell-ide-engine/builds/618340235
I don't know why the Test stage expands differently than the other stages in this build.

To me this documentation implies it would work for something other than an environment variable: https://docs.travis-ci.com/user/build-stages/#build-stages-and-build-matrix-expansion